modules/imgcodecs/src/grfmt_jpeg2000.cpp

```diff
@@ -476,6 +477,17 @@ bool  Jpeg2KEncoder::write( const Mat& _img, const std::vector<int>& )
     if( channels > 3 || channels < 1 )
         return false;
 
+    double target_compression_rate = 1.0;
+    for( size_t i = 0; i < params.size(); i += 2 )
+    {
+        switch(params[i])
+        {
+        case cv::IMWRITE_JPEG2000_COMPRESSION:
+            target_compression_rate /= params[i+1];
```

Comment:
handle the case when `params[i+1] <= 0`

Reply:
Very valid point!
Concerning the definition of "too big". I guess if 1/compression is numerically 0 is likely a good boundary, right?

Reply:
well, maybe not. Maybe 100 (or even 10) will already be such a big number that the produced image will have very low quality. For example, Photoshop uses the compression quality in from 0 to 12, some other editors use percentage (100% is the best quality, %50 is so-so, closer to 0% the image degrades very much). I think, it would be useful to have some guidance - how this quality parameter maps to the percentage or to the Photoshop scale.

Reply:
Hm, a value of 100 would mean a target rate of 1%, i.e. depending on the image it might still be a possible value.
I would actually refrain from imposing an artificial upper limit there is no generally applicable limit for this parameter.
Maybe, one should point out in the documentation that it is a target, not a guaranteed value?